### PR TITLE
upgrade to discordx and discord.js@latest

### PR DESCRIPTION
**Pull request by vijayymmeena**
### Files with type errors
- src/db/**
- src/commands/moderation
- src/guards/config/**
- src/helpers/send-sytem-message

note: remove any usage from commands/config.ts at SlashChoice

### CODE REVIEW

File: src/bot.ts [hunk 1]
```diff
@@ -25,16 +25,16 @@ export class Bot {
...
-    userId: string,
-    @Option('message', OptionType.STRING)
+    @SlashOption('user', { type: 'USER', required: true })
+    user: User | GuildMember,
+    @SlashOption('message', { type: 'STRING', required: true })
     message: string,
     interaction: CommandInteraction
   ): Promise<void> {
-    await sendMessageToUser(interaction.client, userId, message);
+    await sendMessageToUser(interaction.client, user.id, message);
```
Comment (AndyClausen) [inline src/bot.ts:34]: Does this mean that discordx resolves users automatically?

File: src/commands/config.ts [hunk 2]
```diff
@@ -36,21 +26,21 @@ export default class Config {
...
 
   @Slash('enable')
   @Guard(IsValidKey)
   async enable(
-    @Choices(
+    @SlashChoice(
       Object.keys(ConfigModel.schema.paths)
         .filter(ConfigClass.isBooleanProp)
-        .reduce<ChoicesType>((obj, key) => {
+        .reduce<any>((obj, key) => {
```
Comment (AndyClausen) [inline src/commands/config.ts:38]: Why is this changed to `any`?

File: src/commands/moderation.ts [hunk 3]
```diff
@@ -19,12 +20,14 @@ import GuardCache from '../types/GuardCache';
 export default abstract class Moderation {
   @Slash('move', { description: 'Move all users from one voice channel to another' })
   async bulkMove(
-    @Option('to', OptionType.CHANNEL, {
+    @SlashOption('to', {
+      type: 'CHANNEL',
       description: 'Channel to move to',
       required: true,
     })
     channelTo: GuildChannel,
```
Comment (AndyClausen) [inline src/commands/moderation.ts:28]: Will this also be resolved automatically? If so, the resolving below should be removed.

File: src/bot.ts (hunk 1 above)
Comment (vijayymmeena) [reply src/bot.ts:34]: yes, if you set option optional then it's

user | guildmember | undefined

File: src/commands/config.ts (hunk 2 above)
Comment (vijayymmeena) [reply src/commands/config.ts:38]: left note for it, this is for you to resolve.

File: src/commands/moderation.ts (hunk 3 above)
Comment (vijayymmeena) [reply src/commands/moderation.ts:28]: yes, there are many improvements like this on overall your project but I focus on migration only. Rest, should be done as per your need or usage.
